src/features/defi/providers/osmosis/components/OsmosisManager/Withdraw/components/Confirm.tsx

```diff
@@ -0,0 +1,279 @@
+import { logger } from 'lib/logger'
+import {
+  selectAssetById,
+  selectBIP44ParamsByAccountId,
+  selectMarketDataById,
+  selectPortfolioCryptoHumanBalanceByFilter,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { OsmosisWithdrawActionType } from '../WithdrawCommon'
+import { WithdrawContext } from '../WithdrawContext'
+
+const moduleLogger = logger.child({
+  namespace: ['Defi', 'Providers', 'Osmosis', 'OsmosisManager', 'Withdraw', 'Confirm'],
+})
+
+type ConfirmProps = { accountId: AccountId | undefined } & StepComponentProps
+
+export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
+  const { state, dispatch: contextDispatch } = useContext(WithdrawContext)
+  const translate = useTranslate()
+  const { query } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId } = query
+  const opportunity = state?.opportunity
+
+  const chainAdapter = getChainAdapterManager().get(chainId) as unknown as osmosis.ChainAdapter
+
+  const feeAsset = useAppSelector(state => selectAssetById(state, osmosisAssetId))
+
+  const underlyingAsset0 = useAppSelector(state =>
+    selectAssetById(state, opportunity?.underlyingAssetIds[0] || ''),
+  )
+  const underlyingAsset1 = useAppSelector(state =>
+    selectAssetById(state, opportunity?.underlyingAssetIds[1] || ''),
+  )
+
+  const feeMarketData = useAppSelector(state => selectMarketDataById(state, osmosisAssetId))
+
+  const accountFilter = useMemo(() => ({ accountId }), [accountId])
+  const bip44Params = useAppSelector(state => selectBIP44ParamsByAccountId(state, accountFilter))
+  const userAddress = useMemo(() => accountId && fromAccountId(accountId).account, [accountId])
+
+  // user info
+  const { state: walletState } = useWallet()
+
+  // notify
+  const toast = useToast()
+
+  const feeAssetBalanceFilter = useMemo(
+    () => ({ assetId: feeAsset?.assetId, accountId: accountId ?? '' }),
+    [accountId, feeAsset?.assetId],
+  )
+  const feeAssetBalance = useAppSelector(s =>
+    selectPortfolioCryptoHumanBalanceByFilter(s, feeAssetBalanceFilter),
+  )
+
+  const handleWithdraw = useCallback(async () => {
+    if (
+      !(
+        contextDispatch &&
+        state &&
+        state.poolData &&
+        userAddress &&
+        walletState &&
+        walletState.wallet &&
+        supportsOsmosis(walletState.wallet) &&
+        opportunity &&
+        chainAdapter &&
+        bip44Params
+      )
+    ) {
+      return
+    }
+    try {
+      contextDispatch({ type: OsmosisWithdrawActionType.SET_LOADING, payload: true })
+
+      const estimatedFees = await chainAdapter.getFeeData({ sendMax: false })
+      const result = await (async () => {
+        const fees = estimatedFees.average as FeeData<CosmosSdkChainId>
+        const {
+          chainSpecific: { gasLimit },
+          txFee,
+        } = fees
+
+        if (!state.poolData || !state.poolData.id || !walletState.wallet) return
```

Comment:
We probably want to move this above not to make a fees estimation if we're not ready to deposit